relayer/application.ts

```diff
@@ -212,25 +276,35 @@ export class RelayerApp<ContextT extends Context> extends EventEmitter {
    * @param vaa
    * @param opts
    */
-  async pushVaaThroughPipeline(vaa: Buffer, opts?: any): Promise<void> {
-    const parsedVaa = wormholeSdk.parseVaa(vaa);
+  private async pushVaaThroughPipeline(
+    vaa: SignedVaa,
+    opts: any,
+  ): Promise<void> {
+    const parsedVaa = parseVaaWithBytes(vaa);
+    const job: RelayJob | undefined = opts.job;
+
     let ctx: Context = {
-      vaa: parsedVaa,
-      vaaBytes: vaa,
-      env: this.env,
-      fetchVaa: this.fetchVaa.bind(this),
-      fetchVaas: this.fetchVaas.bind(this),
-      processVaa: this.processVaa.bind(this),
       config: {
         spyFilters: await this.spyFilters(),
       },
+      env: this.env,
+      fetchVaa: this.fetchVaa.bind(this),
+      fetchVaas: this.fetchVaas.bind(this),
       locals: {},
+      on: this.on.bind(this),
+      processVaa: this.processVaa.bind(this),
+      vaa: parsedVaa,
+      vaaBytes: vaa,
     };
-    Object.assign(ctx, opts);
+    Object.assign(ctx, opts, { storage: { job } });
     try {
-      await this.pipeline?.(ctx, () => {});
+      await this.pipeline?.(ctx, () => {
+        this.emit(RelayerEvents.Completed, parsedVaa, job);
```

Comment:
Less error prone if you emit the event after the await instead of relying on the last middleware to call `next()`

Reply:
great catch ty